src/python/pants/backend/python/target_types_test.py

```diff
@@ -173,33 +184,55 @@ def test_inject_pex_binary_entry_point_dependency(caplog) -> None:
         ],
         target_types=[PexBinary, PythonRequirementLibrary, PythonLibrary],
     )
-    rule_runner.add_to_build_file(
```

Comment:
Rewritten to use `.write_files()`. Only change made is adding `ambiguous_in_another_root.py` file and targets.